CharlieBackend.Business/Services/ScheduleService.cs

```diff
@@ -0,0 +1,169 @@
+                }
+
+                var scheduleEntity = _mapper.Map<Schedule>(scheduleModel);
+
+                if (IsNotValidRepeating(scheduleEntity))
+                {
+                    return Result<ScheduleDto>.Error(ErrorCode.ValidationError, "DayNumber can`t be null");
+                }
+
+                scheduleEntity.StudentGroup = await _unitOfWork.StudentGroupRepository
+                    .GetByIdAsync(scheduleEntity.StudentGroupId);
+
+                if(scheduleEntity.StudentGroup == null)
+                {
+                    return Result<ScheduleDto>.Error(ErrorCode.NotFound, "StudentGroupId is not valid");         
+                }
+
+                _unitOfWork.ScheduleRepository.Add(scheduleEntity);
+             
+                await _unitOfWork.CommitAsync();
+
+                return Result<ScheduleDto>.Success(_mapper.Map<ScheduleDto>(scheduleEntity));
+            }
+            catch
+            {
+                _unitOfWork.Rollback();
+
+                return Result<ScheduleDto>.Error(ErrorCode.InternalServerError, "Internal error");
+            }
+        }
+
+        public async Task<Result<ScheduleDto>> DeleteScheduleByIdAsync(long id)
+        {
+            var scheduleEntity = await _unitOfWork.ScheduleRepository.GetByIdAsync(id);
+
+            if(scheduleEntity != null)
+            {
+                var mappedSchedule = _mapper.Map<ScheduleDto>(scheduleEntity);
+                await _unitOfWork.ScheduleRepository.DeleteAsync(id);
+
+                await _unitOfWork.CommitAsync();
+
+                return Result<ScheduleDto>.Success(mappedSchedule);
+            }
+
+            return Result<ScheduleDto>.Error(ErrorCode.NotFound, "scheduleId is not valid");
+        }
+
+        public async Task<IList<ScheduleDto>> GetAllSchedulesAsync()
+        {
+            var scheduleEntities = await _unitOfWork.ScheduleRepository.GetAllAsync();
+
+            return _mapper.Map<IList<Schedule>, IList<ScheduleDto>>(scheduleEntities);
+        }
+
+        public async Task<Result<ScheduleDto>> GetScheduleByIdAsync(long id)
+        {
+            var scheduleEntity = await _unitOfWork.ScheduleRepository.GetByIdAsync(id);
+
+            return scheduleEntity == null ? 
+                Result<ScheduleDto>.Error(ErrorCode.NotFound, "Schedule id is not valid") :
+                Result<ScheduleDto>.Success(_mapper.Map<ScheduleDto>(scheduleEntity));
+        }
+
+        public async Task<IList<ScheduleDto>> GetSchedulesByStudentGroupIdAsync(long id)
+        {
+            var groupEntity = await _unitOfWork.StudentGroupRepository.GetByIdAsync(id);
+
+            if(groupEntity == null)
+            {
+               return null;
+            }
+            var schedulesOfGroup = await _unitOfWork.ScheduleRepository.GetSchedulesByStudentGroupIdAsync(id);
+
+            return _mapper.Map<IList<Schedule>, IList<ScheduleDto>>(schedulesOfGroup);
```

Comment:
Doesn't that work with only the destination type?
```suggestion
            return _mapper.Map<IList<ScheduleDto>>(schedulesOfGroup);
```

Reply:
Done